chain/src/main/scala/org/bitcoins/chain/blockchain/ChainHandler.scala

```diff
@@ -429,12 +431,16 @@ case class ChainHandler(
         } else {
           val startHeight = end - (batchSize - 1)
           val endHeight = end
-          val newAcc: Future[Vector[DoubleSha256DigestBE]] = for {
+          val newAcc = for {
             compactFilterDbs <- filterDAO.getBetweenHeights(startHeight,
                                                             endHeight)
-            groupSize = calcGroupSize(compactFilterDbs.size)
-            grouped = compactFilterDbs.grouped(groupSize)
-            filtered <- findMatches(grouped)
+            filtered <- if (compactFilterDbs.isEmpty)
```

Comment:
Do we really need this check here? Can't we just delegate this to the else clause and have `findMatches()` not work on a `Vector.empty`?

Reply:
We do. `Iterable.grouped()` throws if group size is less than 1. Also there's no reason to allocate so many entities for an empty DB response.